include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -56,8 +56,12 @@ __pattern_walk1(__hetero_tag<_BackendTag>, _ExecutionPolicy&& __exec, _ForwardIt
         oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::read_write, _ForwardIterator>();
     auto __buf = __keep(__first, __last);
 
+    auto __view = __buf.all_view();
```

Comment:
Should we really extract this into local variable? Will we have some profit?

Reply:
Initially, I did this to remove the duplicate `all_view()`s that arise but this actually seems to lead to more code looking at the diff. All of these added local view variables have been reverted in the PR.